aws/resource_aws_dynamodb_table.go

```diff
@@ -840,50 +845,94 @@ func flattenAwsDynamoDbTableResource(d *schema.ResourceData, meta interface{}, t
 func resourceAwsDynamoDbTableDelete(d *schema.ResourceData, meta interface{}) error {
 	dynamodbconn := meta.(*AWSClient).dynamodbconn
 
-	if err := waitForTableToBeActive(d.Id(), meta); err != nil {
-		return errwrap.Wrapf("Error waiting for Dynamo DB Table update: {{err}}", err)
-	}
-
 	log.Printf("[DEBUG] DynamoDB delete table: %s", d.Id())
 
-	_, err := dynamodbconn.DeleteTable(&dynamodb.DeleteTableInput{
-		TableName: aws.String(d.Id()),
+	err := resource.Retry(d.Timeout(schema.TimeoutDelete), func() *resource.RetryError {
+		_, err := dynamodbconn.DeleteTable(&dynamodb.DeleteTableInput{
+			TableName: aws.String(d.Id()),
+		})
+		if err != nil {
+			// Table is already deleted
+			if isAWSErr(err, dynamodb.ErrCodeResourceNotFoundException, "Requested resource not found: Table: ") {
+				return nil
+			}
+			// This logic handles multiple scenarios in the DynamoDB API:
+			// 1. Updating a table immediately before deletion may return:
+			//    ResourceInUseException: Attempt to change a resource which is still in use: Table is being updated:
+			// 2. Removing a table from a DynamoDB global table may return:
+			//    ResourceInUseException: Attempt to change a resource which is still in use: Table is being deleted:
+			if isAWSErr(err, dynamodb.ErrCodeResourceInUseException, "") {
+				return resource.RetryableError(err)
+			}
+			// Unknown error
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
 	})
+
+	// check error from retry
 	if err != nil {
 		return err
 	}
 
-	params := &dynamodb.DescribeTableInput{
+	log.Println("[INFO] Waiting for DynamoDB Table to be destroyed")
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			dynamodb.TableStatusActive,
+			dynamodb.TableStatusCreating,
```

Comment:
Could this be a pending state when destroying?

Reply:
Same logic about edge-cases as mentioned before. Let me know if you think I'm overthinking this.